util/chplenv/chpl_llvm.py

```diff
@@ -365,37 +371,30 @@ def get_llvm_clang_command_name(lang):
     else:
         return 'clang'
 
-@memoize
-def get_system_llvm_clang(lang):
-    clang_name = get_llvm_clang_command_name(lang)
+def get_possible_clang_installation_directories():
+    paths = []
     bindir = get_system_llvm_config_bindir()
-    clang = ''
     if bindir:
-        clang = os.path.join(bindir, clang_name)
-
-        if not os.path.exists(clang):
-            # try /usr/bin/clang-<version> or /usr/bin/clang
-            # since some OSes use that for the clang package
-            paths = [ ]
-
-            usr_bin = "/usr/bin"
-            llvm_config = find_system_llvm_config()
-            llvm_version, ignored_err = check_llvm_config(llvm_config)
+        paths.append(bindir)
+    return paths + os.environ["PATH"].split(":")
 
-            paths.append(os.path.join(usr_bin, clang_name + "-" + llvm_version))
-            paths.append(os.path.join(usr_bin, clang_name))
-
-            for clang2 in paths:
-                if os.path.exists(clang2):
-                    # check that clang --version matches llvm-config --version
-                    clangv = run_command([clang2, '--version']).strip()
-                    llvmv = run_command([llvm_config, '--version']).strip()
+def get_possible_clang_executables(lang):
+    clang_name = get_llvm_clang_command_name(lang)
+    for folder in get_possible_clang_installation_directories():
+        for file in os.listdir(folder):
+            if clang_name in file:
```

Comment:
I think we need to change to checking that `file` is exactly `clang_name` or `clang_name + "-" + llvm_version`. The issue is that, on (for example) my Ubuntu system, /usr/bin has a variety of commands in `/usr/bin` containing `clang` as a substring but that aren't useful or relevant here. In particular `clang-cpp-13` `clangd` `clangd-15` are some examples.

Reply:
Good catch! Should be fixed by the latest commit. Does it look reasonable to you?

Reply:
Yes. I haven't started testing it yet, but it looks better. I think it'll be the first `import itertools` in util/chplenv (I would have just written a nested `for` loop). I think that is OK, but just FYI, this script needs to work with Python 2.7 as well as Python 3.

Reply:
Ah, I didn't think about Python 2 (since it EOL by now ;) ). If it fails, I can switch to nested loops.